app/code/Magento/CatalogGraphQl/Model/Resolver/Products/DataProvider/Product/CollectionProcessor/PriceDataProcessor.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Products\DataProvider\Product\CollectionProcessor;
+
+use Exception;
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
+use Magento\CatalogGraphQl\Model\Resolver\Products\DataProvider\Product\CollectionProcessorInterface;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Api\Data\GroupInterface;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\GraphQl\Model\Query\ContextInterface;
+
+/**
+ * Add price data to product collection results
+ *
+ * @inheritdoc
+ */
+class PriceDataProcessor implements CollectionProcessorInterface
+{
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @param CustomerRepositoryInterface $customerRepository
+     */
+    public function __construct(
+        CustomerRepositoryInterface $customerRepository
+    ) {
+        $this->customerRepository = $customerRepository;
+    }
+
+    /**
+     * Process to add price data to product collection.
+     *
+     * @param Collection $collection
+     * @param SearchCriteriaInterface $searchCriteria
+     * @param array $attributeNames
+     * @param ContextInterface|null $context
+     * @return Collection
+     */
+    public function process(
+        Collection $collection,
+        SearchCriteriaInterface $searchCriteria,
+        array $attributeNames,
+        ContextInterface $context = null
+    ): Collection {
+        /**
+         * The $customerGroupId which is used to add price data to collection.
+         * Falls back to zero which represents the "Not Logged In" customer group.
+         */
+        try {
+            if ($context
+                && $context->getExtensionAttributes()->getIsCustomer() === true
+            ) {
+                $customer = $this->customerRepository->getById(
```

Comment:
I think that this is a possible performance degradation point. Can we try to check the customer groupId by using the current customer session? `\Magento\Customer\Model\Session::getCustomerGroupId`

Reply:
Hello @coderimus 
Thank you for your review.
I completely agree with you, but it seems to me that this is the problem of the context object of GraphQl and we should resolve it separately.